src/dali/plugin/input/rest/kraken.py

```diff
@@ -122,10 +122,31 @@ def _initialize_client(self) -> kraken:
 
     def _initialize_markets(self) -> None:
         self._client.load_markets()
-        self._client.markets_by_id.update({"BSVUSD": {_ID: "BSVUSD", _BASE_ID: "BSV", _BASE: "BSV", _QUOTE: "USD"}})
+        markets_by_ids: Dict[str, List[Dict[str, str]]] = self._client.markets_by_id  # type: ignore
+
+        markets_by_ids.update({"BSVUSD": [{_ID: "BSVUSD", _BASE_ID: "BSV", _BASE: "BSV", _QUOTE: "USD"}]})
+
+        for markets in self._client.markets_by_id.values():
+            if not isinstance(markets, list):  # type: ignore
+                exc_str = f"Expected List from Kraken CCXT Exchange, got {type(markets)} instead. " \
+                          f"Incompatible CCXT library - make sure to follow Dali setup instructions " \
+                          f"to install appropriate versions of dependencies."
+                raise RP2RuntimeError(exc_str)
+
+            for market in markets:  # type: ignore
+                base_id: str = market[_BASE_ID]
+                if base_id in self.base_id_to_base and \
+                    market[_BASE] != self.base_id_to_base[base_id]:
+                    self.__logger.warning(
```

Comment:
Can you clarify in a comment how this happens? I'm also trying to understand how bad this is: is it just a warning (recoverable) or is it a RP2RuntimeError?

Reply:
I've never seen it happen but because `ccxt` passes back a list I didn't want it to go unnoticed. I'm assuming `ccxt` may have an exchange that has multiple markets for a pair. It's not clear what prompted them to implement it this way but in case kraken did do it I added the warning to log it. Would it make more sense to make it an exception? The other option is to remove it as it may be the result of another exchange.

Reply:
I would side with it causing an issue with computation. Incorrectly categorizing the asset would affect the cost basis and taxable event matching. Having said that I'll make sure to raise an exception and add a comment explaining how this could happen.

Reply:
Sounds like a good plan!